### DISCUSSION TITLE
General advice on using pgvector with marten

### DISCUSSION THREAD

**Discussion by wmeints** [JasperFx/marten#3429] [General] [answered]
Hi, I'm building an application that has a mix of storage: I have regular relational data/documents, and I have a piece data that uses pgvector types in postgres. I would love to use Marten for my application, but I figured I can't because I have a table using vector.

From what I understand, I can use marten to store most of my data. But I need to drop down to SQL for the vector related stuff. Or is there another way?

**Comment by wmeints** [comment]
I searched around in the docs and maybe have a solution:

- I need a document that has a float[] data type for the vector.
- I create a duplicated column of type vector(1536)
- When I want to search for cosine distance, I use [Advanced SQL](https://martendb.io/documents/querying/advanced-sql.html)

Still, I'm curious if this is something that could work.

**Comment by mysticmind** [comment]
Currently, there is no native support for pgvector vector type and the operations surrounding it. So what you are doing to use advanced sql is the way to go for now. I will take a look on support of it and keep you posted. But it is not going to be too quick.

**Comment by wmeints** [comment]
I can live with the advanced sql option in this case. The scenario I'm working is rather niche I think. Most people would use a regular vector store. But I want the niceties of Marten.

**Comment by mysticmind** [comment, marked as answer]
PgVector support is available now, https://martendb.io/postgres/pgvector.html
